### PR TITLE
feat[ci]: parallelize integration, example, and sync tests

### PR DESCRIPTION
**Pull request by annieke**
**Description**
Make integration testing more efficient by parallelizing the 3 test suites.

**Additional context**
following up on https://github.com/ethereum-optimism/optimism/pull/913

**Metadata**
- OP-810

### CODE REVIEW

File: ops/scripts/save-ci.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "deployer"
+    "data-transport-layer"
+    "batch-submitter"
+    "message-relayer"
+    "integration-tests"
+    "l2geth"
+    "hardhat"
+)
+
+docker images
```
Comment (annieke) [inline ops/scripts/save-ci.sh:13]: note: remove this after debugging

**Review by gakonst — CHANGES_REQUESTED** [review]
The work flows still take many minutes to run because the load-images step is not finding the tarballs

**Review by gakonst — CHANGES_REQUESTED** [review]
This PR does what we want, but it seems to be making things slower due to having to save/upload & download/load the images...If this does not improve our CI times let's hold on it. I've opened #989 to enable the sync tests on CI.

**Comment by snario** [comment]
Curious if you considered using the caching strategy Docker proposes [here](https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md#github-cache)?

**Comment by annieke** [comment]
@snario just poked around some docs and i think for us to share images between jobs (necessary for parallelizing the tests), the uploading artifacts part is still necessary, and the caching just helps with the build part: https://github.com/docker/build-push-action/issues/225 let me know if your investigations yield any further results! going to mark this as draft until then

**Pull request closed without merging**